frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx

```diff
@@ -109,9 +109,9 @@ export function PropertyValue({
 
     const commonInputProps = {
         onSearch: (newInput: string) => {
-            setInput(newInput)
```

Comment:
The change needs to happen here instead of [line209](https://github.com/lawalbolaji/posthog/blob/9f839d3ae30363c4faa73e16f5cc8763e65cf50b/frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx#L209) for 2 reasons:

- The search event fires **after** the change event, and because react will batch the state updates, any state modification in the onChange handler is basically useless.
- I couldn't find a scenario where user input did not fire the search event, and even when users select an autocomplete suggestion, the select event is handled by the onSelect handler.

I would go as far as ditching the onChange handler but the docs insist on using it for controlled inputs... even though onSelect and onSearch are called right after, depending on context.

Reply:
A somewhat annoying side effect of this solution is that if I have a space in the filter (in between letters) I can't type it anymore, so for example if I wanted to type a[space]c, I'll need to type `ac` and then add a space in between, as the `trim()` doesn't let me input `a[space]`.

But, this problem is better than the current one, so I'll merge this as is, but feel free to fix this in a follow up :)